hack/tests/scaffolding/scaffold-memcached.go

```diff
@@ -271,6 +272,15 @@ func insertGoModReplace(repo, path, sha string) ([]byte, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to read go.mod")
 	}
+	// Remove all replace lines in go.mod.
+	replaceRe := regexp.MustCompile(fmt.Sprintf("(replace )?%s =>.+", repo))
```

Comment:
This looks like the regex will just remove everything after a `replace github.com/operator-framework/operator-sdk =>` or `github.com/operator-framework/operator-sdk =>`. Is that correct, and can we always assume that our operator-sdk replace will be the last line in the `go.mod`? Also, if our replace is in a replace block like:

```
replace (
        github.com/coreos/prometheus-operator => github.com/coreos/prometheus-operator v0.29.0
        github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v0.9.0
)
```

which is a case that we are considering based on the regex, then wouldn't there be a missing closing parentheses in that case, which would break the `go.mod` file?

Reply:
I think this would be the regex that captures just the replace line itself (along with whitespace before and after)
```
[[:space:]]*(replace )?github.com/operator-framework/operator-sdk => [^[:space:]]*( v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+)?[[:space:]]*
```

Reply:
The current regex works with your example. The closing parenthesis is not removed. Try it out [here](https://regex-golang.appspot.com/assets/html/index.html). We also do not know what the bit after `=>` looks like; it could be a dir path or package path with a version or branch or commit.

Reply:
I thought that the `.` symbol matches any character, but now looking at the documentation, it is briefly mentioned that `.` optionally matches `\n` only if you set a certain flag. So `.+` should stop matching at the end of the line, so that works I guess.